code/datums/components/faction_granter.dm

```diff
@@ -0,0 +1,58 @@
+#define DEFAULT_GRANT_MESSAGE "You have become friends with "
+
+/**
+ * ## faction granter component!
+ *
+ * component attached to items, lets them be used in hand once to add yourself to a certain faction
+ * one good example is the chaplain plushie that grants you the carp faction, making you friendly with them.
+ */
+/datum/component/faction_granter
+	dupe_mode = COMPONENT_DUPE_ALLOWED
+	///whichever faction the parent adds upon using in hand
+	var/faction_to_grant
+	///whether you need to be holy to get the faction.
+	var/holy_role_required
+	///message given when granting the faction.
+	var/grant_message
+	///boolean on whether it has been used
+	var/used = FALSE
+
+/datum/component/faction_granter/Initialize(faction_to_grant, holy_role_required = NONE, grant_message = DEFAULT_GRANT_MESSAGE)
```

Comment:
just let it default to null, checking to see if it's this particular string is kinda strange

Reply:
this makes it much much harder for admins to add :^(

Reply:
why? have it default to null, if it's null set it to the full string + the faction. hell if you do that you don't need the define anymore, since it'll only ever be used in one place